libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -711,6 +712,11 @@ void AP_DDS_Client::on_topic(uxrSession* uxr_session, uxrObjectId object_id, uin
     (void) request_id;
     (void) stream_id;
     (void) length;
+    auto *external_control = AP::externalcontrol();
+    if (!external_control->is_enabled()) {
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s Command rejected: External Control Disabled", msg_prefix);
```

Comment:
what happens if a companion computer script is running high rate (eg. 200Hz) DDS control when you use the switch to disable external control? This looks like it will flood the telemetry link, and the user will lose all situational awareness. I think you need to rate limit this message unless I've misunderstood something